### DISCUSSION TITLE
Migration different file format

### DISCUSSION THREAD

**Discussion by abhishek-kumar-96** [ariga/atlas#2900] [Q&A]
I want to generate different file format of the migration files

In the doc its written this:

> By default, migration files are named with the following format {{ now }}_{{ name }}.sql. If you wish to use a different file format, use the format query parameter in the directory URL.

But doesn't have any example or explains where this `format` will go i tried many combinations by putting in `--dir` but it was futile, can anyone pls help how can i achieve this

This is my sample `atlas.hcl`

```
data "external_schema" "sqlalchemy" {
    program = [
        "python3",
        "load_models.py"
    ]
}

env "sqlalchemy" {
  src = data.external_schema.sqlalchemy.url
  dev = "docker://postgres/16/dev?search_path=public"
  migration {
    dir = "file://app_server/core/migrations/versions"
  }
  format {
    migrate {
      diff = "{{ sql . \"  \" }}"
    }
  }
}
```

**Comment by masseelch** [comment]
Hey. `format` in this case is a query parameter in the directory url, specifying the migration tool format you want to use, e.g. for goose:

`atlas migrate diff --dir file://migrations?format=goose --to file://schema.hcl` will generate the goose compatible file format.
